### PR TITLE
#624: Use currency from backend in price formatter

### PR DESCRIPTION
**Pull request by bonkey**
- Refactored code to use Money instead of MoneyAmount and Price
- Refactored shipping price to be dynamic based on totalPrice

### CODE REVIEW

**Comment by bonkey** [comment]
👍

**Comment by bonkey** [comment]
👍

File: AtlasSDK/AtlasSDK/Models/Price.swift [hunk 1]
```diff
@@ -1,37 +0,0 @@
-//
-//  Copyright © 2016 Zalando SE. All rights reserved.
-//
-
-import Foundation
-
-public struct Price {
```
Comment (hani-ibrahim) [inline AtlasSDK/AtlasSDK/Models/Price.swift]: Nice 👍

File: AtlasUI/AtlasUI/Utils/Localizer.swift [hunk 2]
```diff
@@ -106,8 +107,12 @@ extension Localizer {
         return shared.format(string: string, formatArguments: formatArguments)
     }
 
-    static func format(price: NSNumber) -> String? {
-        return shared.format(price: price)
+    static func format(price: NSNumber, withCurrency currencySymbol: String? = nil) -> String? {
```
Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/Utils/Localizer.swift]: why we have two method with the same signature but one is static and one is not?

**Comment by hani-ibrahim** [comment]
👍

File: AtlasUI/AtlasUI/Utils/Localizer.swift (hunk 2 above)
Comment (bonkey) [inline AtlasUI/AtlasUI/Utils/Localizer.swift]: Just to not write `shared` in `Localizer.shared.format(...` everytime

Comment (hani-ibrahim) [inline AtlasUI/AtlasUI/Utils/Localizer.swift]: then may be remove the non static one and add the logic directly in the static one?

**Comment by bonkey** [comment]
👍

File: AtlasUI/AtlasUI/Utils/Localizer.swift (hunk 2 above)
Comment (bonkey) [inline AtlasUI/AtlasUI/Utils/Localizer.swift]: I don't want to castrate real class with it, these statics are just syntax sugar

**Comment by hani-ibrahim** [comment]
👍

**Pull request merged**